src/drivers/imu/adis16477/ADIS16477.cpp

```diff
@@ -71,86 +70,37 @@ static constexpr uint16_t PROD_ID_ADIS16477 = 0x405D; /* ADIS16477 Identificatio
 
 static constexpr int T_STALL = 16;
 
-#define GYROINITIALSENSITIVITY		250
-#define ACCELINITIALSENSITIVITY		(1.0f / 1200.0f)
-#define ACCELDYNAMICRANGE		18.0f
+static constexpr int ADIS16477_DEFAULT_RATE = 1000;
 
 using namespace time_literals;
 
-ADIS16477::ADIS16477(int bus, const char *path_accel, const char *path_gyro, uint32_t device, enum Rotation rotation) :
-	SPI("ADIS16477", path_accel, bus, device, SPIDEV_MODE3, 1000000),
+ADIS16477::ADIS16477(int bus, uint32_t device, enum Rotation rotation) :
```

Comment:
Hi @dagar.  I think enforcing constness here would be a great thing:
```
ADIS16477::ADIS16477(const int bus, const uint32_t device, const enum Rotation rotation) :
```
and providing defaults in the declaration would be a good thing as well!

Reply:
Done.